lib/galaxy/model/store/__init__.py

```diff
@@ -1375,23 +1257,140 @@ def record_associated_jobs(obj):
                 output_dataset_mapping[name].append(self.exported_key(dataset))
             jobs_attrs.append({"id": job_id, 'output_dataset_mapping': output_dataset_mapping})
 
-        icjs_attrs = []
-        for icj in implicit_collection_jobs_dict.values():
-            icj_attrs = icj.serialize(self.security, self.serialization_options)
-            icjs_attrs.append(icj_attrs)
+        if not self.serialization_options.for_edit:
```

Comment:
That if looks iffy right? Maybe there should a TODO here and options used more clearly? I tried to make this code pretty general and it seems to be becoming more metadata specific slowly instead moving the other direction. ``for_edit`` doesn't really naturally seem to imply serialize job data. I feel like it might be better to add options for including job data and set at a different level.

Reply:
Do you think https://github.com/galaxyproject/galaxy/pull/11815/commits/e7e3b00f75cfdbfbe9d1207e2f310687d508dd9e looks better ?